src/libraries/System.Console/src/System/IO/StdInReader.cs

```diff
@@ -367,7 +378,7 @@ internal bool MapBufferToConsoleKey(out ConsoleKey key, out char ch, out bool is
         /// not work, we simply return the char associated with that
         /// key with ConsoleKey set to default value.
         /// </summary>
-        public unsafe ConsoleKeyInfo ReadKey(out bool previouslyProcessed)
+        private unsafe ConsoleKeyInfo ReadKeyCore(bool convertCrToLf, out bool previouslyProcessed)
         {
             // Order of reading:
             // 1. A read should first consult _availableKeys, as this contains input that has already been both read from stdin and processed into ConsoleKeyInfos.
```

Comment:
When we pop from `_availableKeys` we should also apply the conversion.

Reply:
That's a good point. I ended up removing the conversion completely, since a) it's not really necessary in the `ReadLine()` and b) it could leak to the buffer via `_tmpKeys`.

Reply:
Actually, turns out I need to still add this to make the Read() and Peek() methods happy. Relevant change: https://github.com/dotnet/runtime/pull/37491/commits/69d6960bffc25895e4ce56905858301085e0b184